code/game/machinery/machinery.dm

```diff
@@ -479,3 +530,122 @@ Class Procs:
 		emp_act(2)
 	else
 		ex_act(2)
+
+/obj/machinery/proc/printer_can_print(alert = FALSE)
+	if(max_printjobs && printer_spooler.len >= max_printjobs)
+		if(alert)
+			printer_memory_full()
+		return FALSE
+	else
+		return TRUE
+
+/obj/machinery/proc/printer_memory_full()
+	if(prob(20))
+		src.visible_message("<span class='danger'>The [src] printer beeps with a message: PC LOAD LETTER</span>")
```

Comment:
in general our coding standard is to avoid specifying src unless it's actually necessary, so just use visible_message